ansible/roles/test/files/ptftests/py3/advanced-reboot.py

```diff
@@ -1870,23 +1869,24 @@ def start_sniffer(self, pcap_path, tcpdump_filter, timeout):
         """
         self.tcpdump_data_ifaces = [
             iface for iface in scapyall.get_if_list() if iface.startswith('eth')]
-        processes_list = []
+        process_args = ['dumpcap', '-w', pcap_path, '-f', tcpdump_filter, '-Z', 'none', '-s', '1514', '-t']
         for iface in self.tcpdump_data_ifaces:
-            iface_pcap_path = '{}_{}'.format(pcap_path, iface)
-            process = subprocess.Popen(
-                ['tcpdump', '-i', iface, tcpdump_filter, '-w', iface_pcap_path],
-                stdout=subprocess.DEVNULL,
-                stderr=subprocess.PIPE,
-                text=True
-            )
-            self.log('Tcpdump sniffer starting on iface: {}'.format(iface))
-            processes_list.append(process)
-
-        for proc in processes_list:
-            while True:
-                line = proc.stderr.readline()
-                if not line or 'listening on' in line:
-                    break
+            process_args += ['-i', iface]
+
+        process = subprocess.Popen(process_args, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+        self.log('Dumpcap sniffer process started')
+
+        pcap_existence_check_limit = 10
+        pcap_existence_check_count = 0
+        while not os.path.exists(pcap_path) and pcap_existence_check_count < pcap_existence_check_limit:
```

Comment:
This is 10s timeout? Will that really be enough? Will the file be generated as soon as 1st packet is captured?

We start sniffer before sender. So there is a chance that sniffer won't capture anything before sender is started, and lead to terminate called prematurely?

Reply:
The file is created before any packets are captured; it's created when dumpcap is ready to capture packets. I think if we do see cases where it takes more than 10 seconds to start dumpcap, then that is an issue we need to investigate to see what is slowing things down.